vitalytaranov/surf-curve-be
# Task 9

- **product-service service API endpoint URL:**  https://f6hqzu6u08.execute-api.eu-west-1.amazonaws.com/dev/products
- **CART service API endpoint URL:**  https://wwdllantpg.execute-api.eu-west-1.amazonaws.com/dev/api
- **bff-service service URL:** http://vitalytaranov-bff-api-development.eu-west-1.elasticbeanstalk.com

Examples of calling endpoints:
http://vitalytaranov-bff-api-development.eu-west-1.elasticbeanstalk.com/products
http://vitalytaranov-bff-api-development.eu-west-1.elasticbeanstalk.com/cart/profile/cart

Evaluation criteria:
✅  **3** - A working and correct express application should be in the bff-service folder.
✅  **5** - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service. The response from the bff-service should be the same as if the recipient service was called directly.
✅  **1** - Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes.
How to test:
Get products list
Create new product
Get products list - result shouldn’t have new product
Wait more than 2 minutes
Get products list - result should have new product
✅  **1** - Use NestJS to create bff-service instead of express

expected score 7 out of 7
